bin/run.sh

```diff
@@ -1,24 +1,60 @@
-#! /bin/sh
+#!/bin/bash
 set -e
 
-# If arguments not provided, print usage and exit
-if [ -z "$1" ] || [ -z "$2" ] || [ -z "$3" ]; then
-    echo "usage: run.sh exercise-slug ./relative/path/to/solution/folder/ ./relative/path/to/output/directory/"
+GREEN="\033[0;32m"
+ORANGE="\033[0;33m"
+NC="\033[0m"
+
+printHelpAndExit() {
+    echo -e "Usage: run.sh ${GREEN}-s${NC} exercise-slug ${GREEN}-i${NC} /solution ${GREEN}-o${NC} /output"
+    echo
+    echo -e "${ORANGE}PARAMETERS${NC}"
+    echo -e "${GREEN}   -s, --slug${NC}		Name of exercise slug"
+    echo -e "${GREEN}   -i, --input${NC}		Absolute path to solution folder"
+    echo -e "${GREEN}   -o, --output${NC}		Absolute path to output directory (For result.json report)"
+    echo
     exit 1
-fi
+}
 
-SLUG="$1"
-INPUT_DIR="$2"
-OUTPUT_DIR="$3"
+if [ $# -eq 0 ]; then
+    printHelpAndExit
+fi
 
 BASEDIR=$(dirname "$0")
 
-# echo "$SLUG: testing..."
-# echo "$1"
-# echo "$2"
-# echo "$3"
+#Parse keys
+while(( "$#" )); do
+    case "$1" in
+        -s|--slug)
+            SLUG="$2"
+            shift
+            shift
+            ;;
+        -i|--input)
+            INPUT_DIR="$2"
+            shift
+            shift
+            ;;
+        -o|--output)
+            OUTPUT_DIR="$2"
+            shift
+            shift
+            ;;
+        *) # preserve positional arguments
+            PARAMS="$PARAMS $1"
+            shift
+            ;;
+    esac
+done
+
+# Print settings
+# echo -e "${ORANGE}OPTIONS${NC}"
+# echo -e "${GREEN}SLUG${NC}    = ${SLUG}"
+# echo -e "${GREEN}INPUT${NC}   = ${INPUT_DIR}"
+# echo -e "${GREEN}OUTPUT${NC}  = ${OUTPUT_DIR}"
 # echo "-------------"
-RUNALL=true "${BASEDIR}"/TestRunner --slug "${SLUG}" --solution-directory "${INPUT_DIR}/${SLUG}" --output-directory "${OUTPUT_DIR}" --swift-location $(which swift) --build-directory "/tmp/"
+
+RUNALL=true "${BASEDIR}"/TestRunner --slug "${SLUG}" --solution-directory "${INPUT_DIR}" --output-directory "${OUTPUT_DIR}" --swift-location $(which swift) --build-directory "/tmp"
```

Comment:
I've removed the `/${SLUG}` argument from the solution directory as the solution directory should be the argument that is passed in.